lib/rust/flexer/src/group.rs

```diff
@@ -10,6 +10,155 @@ pub mod rule;
 
 
 
+// ================
+// === Registry ===
+// ================
+
+/// The group Registry is a container for [`Group`]s in the flexer implementation.
+///
+/// It allows groups to contain associations between themselves, and also implements useful
+/// conversions for groups.
+#[derive(Clone,Debug,Default)]
+pub struct Registry {
+    /// The groups defined for the lexer.
+    groups: Vec<Group>
+}
+
+impl Registry {
+    /// Defines a new group of rules for the lexer with the specified `name` and `parent`.
+    ///
+    /// It returns the identifier of the newly-created group.
+    pub fn define_group(&mut self, name:String, parent_index:Option<usize>) -> Identifier {
```

Comment:
As I've written in 2 previous reviews, use `impl Into<String>` here - then the usage of this API will be nicer. Please change that in all constructor-like functions.